File: op 2.5 py/op 2.5 py/func.py
from TBody import TBody
from TBall import TBall
from TPar import TPar

def calculatingNumberOfObjects(number):
	num_of_balls = num_of_pars = number // 2
	if (num_of_balls + num_of_pars != number):
		num_of_balls += 1
	return num_of_balls, num_of_pars

def createListOfBalls(number):
	list_of_balls = []
	print('')
	for i in range (number):
		print("Enter " + str(i + 1) + " ball radius: ")
		radius = float(input())
		ballObject = TBall(radius)
		ballObject.findVolume()
		ballObject.findArea()
		list_of_balls.append(ballObject)
	return list_of_balls

def createListOfPars(number):
	list_of_pars = []
	print('')
	for i in range (number):
		print("Enter " + str(i + 1) + " par height: ")
		height = float(input())
		print("Enter " + str(i + 1) + " par length: ")
		length = float(input())
		print("Enter " + str(i + 1) + " par width: ")
		width = float(input())
		parObject = TPar(height, length, width)
		parObject.findVolume()
		parObject.findArea()
		list_of_pars.append(parObject)
	return list_of_pars

def ballListOutput(number, list_of_balls):
	fullVolume = 0
	print('')
	for i in range (number):
		list_of_balls[i].infoOutput()
		fullVolume += list_of_balls[i].getVolume()
	print("\nFull volume: {!s:<8.8}".format(fullVolume))

def parListOutput(number, list_of_pars):
	fullArea = 0
	print('')
	for i in range (number):
		list_of_pars[i].infoOutput()
		fullArea += list_of_pars[i].getArea()
	print("\nFull area: {!s:<8.8}".format(fullArea))

File: op 2.5/op 2.5/func.cpp
#include "op 2.5.h"
#include <iostream>
using namespace std;

void calculatingNumberOfObjects(int number, int &num_of_balls, int &num_of_pars) {
	num_of_balls = num_of_pars = number / 2;
	if (num_of_balls + num_of_pars != number)
		num_of_balls++;
}

vector<TBall> createVectorOfBalls(int number) {
	vector <TBall> vector_of_balls;
	double radius;
	cout << endl;
	for (int i = 0; i < number; i++) {
		cout << "Enter " << i + 1 << " ball radius: "; cin >> radius;
		TBall ballObject(radius);
		ballObject.findVolume();
		ballObject.findArea();
		vector_of_balls.push_back(ballObject);
	}
	return vector_of_balls;
}

vector<TParallelepiped> createVectorOfPars(int number) {
	vector <TParallelepiped> vector_of_pars;
	double height, length, width;
	cout << endl;
	for (int i = 0; i < number; i++) {
		cout << "Enter " << i + 1 << " par height: "; cin >> height;
		cout << "Enter " << i + 1 << " par length: "; cin >> length;
		cout << "Enter " << i + 1 << " par width: "; cin >> width;
		TParallelepiped parObject(height, length, width);
		parObject.findArea();
		parObject.findVolume();
		vector_of_pars.push_back(parObject);
	}
	return vector_of_pars;
}

void ballVectorOutput(int number, vector <TBall>& ballVector) {
	double fullVolume = 0;
	cout << endl;
	for (int i = 0; i < number; i++) {
		ballVector[i].infoOutput();
		fullVolume += ballVector[i].getVolume();
	}
	cout << endl << "Full volume: " << fullVolume << endl << endl;
}

void parVectorOutput(int number, vector <TParallelepiped>& parVector) {
	double fullArea = 0;
	for (int i = 0; i < number; i++) {
		parVector[i].infoOutput();
		fullArea += parVector[i].getArea();
	}
	cout << endl << "Full area: " << fullArea << endl;
}